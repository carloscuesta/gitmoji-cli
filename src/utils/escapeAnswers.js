// @flow
import { type Answers } from '../commands/commit/prompts'

export const escapeAnswers = (answers: Answers) => {
  const escape = (str: string) =>
    str.replace(/\\/g, '\\\\').replace(/"/g, '\\"')
  answers.title = escape(answers.title)
  answers.message = escape(answers.message)
  if (answers.scope !== undefined) {
    answers.scope = escape(answers.scope)
  }
}

export default escapeAnswers
