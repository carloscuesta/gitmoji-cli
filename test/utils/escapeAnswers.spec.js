import escapeAnswers from '../../src/utils/escapeAnswers'
import * as stubs from './stubs'

describe('escapeAnswers', () => {
  it(`it should escape the answers`, () => {
    expect(escapeAnswers(stubs.answers)).toMatchSnapshot()
  })
})
