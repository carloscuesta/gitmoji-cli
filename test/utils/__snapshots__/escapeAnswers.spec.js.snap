// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`escapeAnswers it should escape the answers 1`] = `
Object {
  "gitmoji": ":zap:",
  "message": "I am a \\\\\\"crazy\\\\\\" message :\\\\\\\\",
  "title": "I am a \\\\\\"crazy\\\\\\" title :\\\\\\\\",
}
`;
